### ISSUE TITLE
[Bug] Exception in event handler chain makes output loading state forever

### ISSUE THREAD

**Issue by huchenlei** [gradio-app/gradio#5351]
### Describe the bug

I am trying implement a gradio call chain that early stops at certain condition. Something like
```python
def bar(i):
  if i:
    raise gr.Error("Early Stop")

buttonA.click(foo, inputs=[i], outputs=[o]).success(bar, inputs=[i], outputs=[o]).success(foobar, outputs=[o])
```

However, this code will make output element forever loading. I would expect the output element to retain the output from `foo` even if `bar` fails.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

Use following code and click `Trigger Failure`. You should observe that the textbox is in loading state forever.

```python
import gradio as gr
import time


def failure():
    raise gr.Error("This should fail!")


def exception():
    raise ValueError("Something went wrong")


def success():
    return True


def warning_fn():
    gr.Warning("This is a warning!")


def info_fn():
    gr.Info("This is some info")


with gr.Blocks() as demo:
    gr.Markdown(
        "Used in E2E tests of success event trigger. The then event covered in chatbot E2E tests."
        " Also testing that the status modals show up."
    )
    with gr.Row():
        result = gr.Textbox(label="Result")
        result_2 = gr.Textbox(label="Consecutive Event")
    with gr.Row():
        success_btn = gr.Button(value="Trigger Success")
        success_btn_2 = gr.Button(value="Trigger Consecutive Success")
        failure_btn = gr.Button(value="Trigger Failure")
        failure_exception = gr.Button(value="Trigger Failure With ValueError")
    with gr.Row():
        trigger_warning = gr.Button(value="Trigger Warning")
        trigger_info = gr.Button(value="Trigger Info")

        success_btn_2.click(success, None, None).success(
            lambda: "First Event Trigered", None, result
        ).success(lambda: "Consecutive Event Triggered", None, result_2)
        success_btn.click(success, None, None).success(
            lambda: "Success event triggered", inputs=None, outputs=result
        )
        failure_btn.click(failure, None, outputs=result).success(
            lambda: "Should not be triggered", inputs=None, outputs=result
        )
        failure_exception.click(exception, None, None)
        trigger_warning.click(warning_fn, None, None)
        trigger_info.click(info_fn, None, None)

demo.queue()

if __name__ == "__main__":
    demo.launch(show_error=True)

```

### System Info

```shell
gradio==3.32.0
```

### Severity

I can work around it

**Comment by abidlabs** [comment]
Hi @huchenlei I don't see that behavior:

<img width="817" alt="image" src="https://github.com/gradio-app/gradio/assets/1778297/24422853-7e02-4cc3-936f-f706a23949c4">

Can you try upgrading to the latest version of `gradio` (3.41.2) and see if that resolves the issue?

**Comment by huchenlei** [comment]
Ah, it eventually timeouts error state. Is there a way to handle expected error?

**Comment by abidlabs** [comment]
> Is there a way to handle expected error?

Can you elaborate? Typically you'd handle the error in your python function itself

**Comment by huchenlei** [comment]
Let me rephrase it a little:

Is there a mechanism to cancel all successive callback in the chain without make all outputs in error state?

Currently the workaround could be like:
```python
def raise_error_if(internal_state: bool):
  if internal_state:
    raise gr.Error("Stop!")

buttonA.click(fn=foo, outputs=[internal_state, *other_outputs]).then(fn=raise_error_if, inputs=[internal_state]).success(...)
```

But this is very awkward.

**Comment by abidlabs** [comment]
The internal state route works. Or you can you the `.success()` function in place of `.then()` which only executes if the first function worked successfully. 

I'll go ahead and close this issue. If you have other general questions (that are not feature requests or bug reports), please ask in our [Forums](https://discuss.huggingface.co/c/gradio/26) or [our Discord](http://discord.gg/feTf9x3ZSB).
